linkerd/app/core/src/proxy/server.rs

```diff
@@ -187,103 +216,75 @@ where
+                let exec =
+                    tokio::executor::DefaultExecutor::current().instrument(info_span!("http1"));
+                let conn = http
+                    .with_executor(exec)
+                    .http1_only(true)
+                    .serve_connection(io, HyperServerSvc::new(svc))
+                    .with_upgrades();
+                Either::A(
+                    drain
+                        .watch(conn, |conn| conn.graceful_shutdown())
+                        .map(|_| ())
+                        .map_err(Into::into),
+                )
             }
 
-            Some(proto) => Either::B({
-                make_http
-                    .make_service(source)
-                    .map_err(|never| match never {})
-                    .and_then(move |http_svc| match proto {
-                        Protocol::Http1 => Either::A({
-                            trace!("detected HTTP/1");
-                            // Enable support for HTTP upgrades (CONNECT and websockets).
-                            let svc = upgrade::Service::new(http_svc, drain.clone());
-                            let exec = tokio::executor::DefaultExecutor::current()
-                                .instrument(info_span!("http1"));
-                            let conn = http
-                                .with_executor(exec)
-                                .http1_only(true)
-                                .serve_connection(io, HyperServerSvc::new(svc))
-                                .with_upgrades();
-                            drain
-                                .watch(conn, |conn| conn.graceful_shutdown())
-                                .map(|_| ())
-                                .map_err(Into::into)
-                        }),
-
-                        Protocol::Http2 => Either::B({
-                            trace!("detected HTTP/2");
-                            let exec = tokio::executor::DefaultExecutor::current()
-                                .instrument(info_span!("h2"));
-                            let conn = http
-                                .with_executor(exec)
-                                .http2_only(true)
-                                .http2_initial_stream_window_size(initial_stream_window_size)
-                                .http2_initial_connection_window_size(initial_conn_window_size)
-                                .serve_connection(io, HyperServerSvc::new(http_svc));
-                            drain
-                                .watch(conn, |conn| conn.graceful_shutdown())
-                                .map(|_| ())
-                                .map_err(Into::into)
-                        }),
-                    })
-            }),
-        });
-
-        Box::new(serve_fut)
+            HttpVersion::H2 => {
+                let exec = tokio::executor::DefaultExecutor::current().instrument(info_span!("h2"));
```

Comment:
Nit/TIOLI: this span is called `h2` while the HTTP 1 span is called `http1`, maybe they should be consistent with each other?
```suggestion
                let exec = tokio::executor::DefaultExecutor::current().instrument(info_span!("http2"));
```

Reply:
This is consistent with how the protocols are referred to in RFCs.